File: SummerLandBackend/Models/ProductVariant.cs
namespace SummerLandBackend.Models;

public class ProductVariant
{
    public int Id { get; set; }

    public int ProductId { get; set; }
    public Product Product { get; set; } = null!;

    public int? SizeId { get; set; }
    public Sizes? Size { get; set; }

    public int? ColourId { get; set; }
    public Colour? Colour { get; set; }

    public BarcodeType BarcodeType { get; set; }

    public string Barcode { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public int LowStockThreshold { get; set; } = 5;

    public ICollection<Inventory> Inventory { get; set; }
        = new List<Inventory>();

    public ICollection<InvoiceItem> InvoiceItems { get; set; }
        = new List<InvoiceItem>();
}

File: summerland/lib/models/product_variant.dart
class ProductVariant {
  final int id;
  final int productId;
  final String modelNumber;
  final String productName;

  final int? sizeId;
  final String? sizeName;

  final int? colourId;
  final String? colourName;

  final int barcodeType;
  final String barcode;

  final double price;
  final int lowStockThreshold;

  final int totalQuantity;
  final bool isLowStock;

  ProductVariant({
    required this.id,
    required this.productId,
    required this.modelNumber,
    required this.productName,
    this.sizeId,
    this.sizeName,
    this.colourId,
    this.colourName,
    required this.barcodeType,
    required this.barcode,
    required this.price,
    required this.lowStockThreshold,
    required this.totalQuantity,
    required this.isLowStock,
  });

  factory ProductVariant.fromJson(
      Map<String, dynamic> json,
      ) {
    return ProductVariant(
      id: json['id'],
      productId: json['productId'],
      modelNumber: json['modelNumber'],
      productName: json['productName'],
      sizeId: json['sizeId'],
      sizeName: json['sizeName'],
      colourId: json['colourId'],
      colourName: json['colourName'],
      barcodeType: json['barcodeType'],
      barcode: json['barcode'],
      price: (json['price'] as num).toDouble(),
      lowStockThreshold:
      json['lowStockThreshold'] ?? 0,
      totalQuantity:
      json['totalQuantity'] ?? 0,
      isLowStock:
      json['isLowStock'] ?? false,
    );
  }
}